pygments/lexers/lisp.py

```diff
@@ -2846,3 +2846,351 @@ class FennelLexer(RegexLexer):
+        'pp', 'prewalk', 'prin', 'prinf', 'print', 'printf',
+        'product', 'propagate', 'put', 'put-in', 'quit',
+        'range', 'reduce', 'reduce2', 'repl', 'require',
+        'resume', 'return', 'reverse', 'reverse!',
+        'run-context', 'sandbox', 'scan-number', 'setdyn',
+        'signal', 'slice', 'slurp', 'some', 'sort', 'sort-by',
+        'sorted', 'sorted-by', 'spit', 'string',
+        'string/ascii-lower', 'string/ascii-upper',
+        'string/bytes', 'string/check-set', 'string/find',
+        'string/find-all', 'string/format', 'string/from-bytes',
+        'string/has-prefix?', 'string/has-suffix?',
+        'string/join', 'string/repeat', 'string/replace',
+        'string/replace-all', 'string/reverse', 'string/slice',
+        'string/split', 'string/trim', 'string/triml',
+        'string/trimr', 'string?', 'struct', 'struct/getproto',
+        'struct/proto-flatten', 'struct/to-table',
+        'struct/with-proto', 'struct?', 'sum', 'symbol',
+        'symbol/slice', 'symbol?', 'table', 'table/clear',
+        'table/clone', 'table/getproto', 'table/new',
+        'table/proto-flatten', 'table/rawget', 'table/setproto',
+        'table/to-struct', 'table/weak', 'table/weak-keys',
+        'table/weak-values', 'table?', 'take', 'take-until',
+        'take-while', 'thaw', 'trace', 'true?', 'truthy?',
+        'tuple', 'tuple/brackets', 'tuple/setmap',
+        'tuple/slice', 'tuple/sourcemap', 'tuple/type',
+        'tuple?', 'type', 'unmarshal', 'untrace', 'update',
+        'update-in', 'values', 'varglobal', 'walk',
+        'warn-compile', 'xprin', 'xprinf', 'xprint', 'xprintf',
+        'yield', 'zero?', 'zipcoll',
+        # obsolete builtin functions
+        'tarray/buffer', 'tarray/copy-bytes', 'tarray/length',
+        'tarray/new', 'tarray/properties', 'tarray/slice',
+        'tarray/swap-bytes', 'thread/close', 'thread/current',
+        'thread/exit', 'thread/new', 'thread/receive',
+        'thread/send'
+    )
+
+    builtin_variables = (
+        'debugger-env', 'default-peg-grammar', 'janet/build',
+        'janet/config-bits', 'janet/version', 'load-image-dict',
+        'make-image-dict', 'math/-inf', 'math/e', 'math/inf',
+        'math/int-max', 'math/int-min', 'math/int32-max',
+        'math/int32-min', 'math/nan', 'math/pi', 'module/cache',
+        'module/loaders', 'module/loading', 'module/paths',
+        'root-env', 'stderr', 'stdin', 'stdout'
+    )
+
+    constants = (
+        'false', 'nil', 'true'
+    )
+
+    # XXX: this form not usable to pass to `suffix=`
+    #_token_end = r'''
+    #  (?=            # followed by one of:
+    #      \s           # whitespace
+    #    | \#           # comment
+    #    | [)\]]        # end delimiters
+    #    | $            # end of file
+    #  )
+    #'''
+
+    # ...so, express it like this
+    _token_end = r'(?=\s|#|[)\]]|$)'
+
+    _first_char = r'[a-zA-Z!$%&*+./<?=>@^_-]'
```

Comment:
I'd prefer to refrain from trying to incorporate such identifiers in this initial PR.

I've been using Janet for some years and looked at a fair bit of Janet code, but AFAICT, almost no one has been using such identifiers.

I looked at what happens currently if things are left as is, and AFAICT, what happens is that the identifier is highlighted / recognized as an error.  It doesn't seem like there are any bad side effects (please share if you know otherwise).  In fact, if one considers that there are various issues [1] with using such identifiers, this might even be seen as a feature.

If at a future point it seems desirable enough for such support,  it seems like it could be added at that point.

---

[1] The following were just a few things that mentioned some issues:

* http://lambda-the-ultimate.org/node/5674
* https://stackoverflow.com/questions/294367/what-are-the-experiences-with-using-unicode-in-identifiers

Reply:
One cost is encouragement of people to use such things without understanding the potential drawbacks.

I doubt most people have given the matter much thought.

I think it would be better if people made informed decisions about these kinds of things and as such having them marked is more likely to lead to investigation.

That aside, AFAICT, not many people are using them and IIUC if there is a demand, they could be added later.

Reply:
I think I'm not comfortable at this time with making this change because the potential consequences and situation are too murky to me.

Perhaps it's best if I close the PR for now and may be it or a similar attempt can be revisited when things are clearer [1].

Thank you so much for your time and detailed review.  I hope that effort gets used at some point in the future.  I'm sorry I didn't understand the situation better before opening this PR.

---

[1] I created a lexer plugin using the [repository you made](https://github.com/pygments/pygments-plugin-scaffolding).  May be that will contribute toward a better understanding of the situation going forward.